forest/src/cli/chain_cmd.rs

```diff
@@ -49,6 +70,81 @@ impl ChainCommands {
                 let cid: Cid = cid.parse().unwrap();
                 print_rpc_res_pretty(chain_get_block((CidJson(cid),)).await);
             }
+            Self::Export {
+                tipset,
+                recent_stateroots,
+                skip_old_messages,
+                output_path,
+            } => {
+                let chain_finality = Policy::default().chain_finality;
+                let recent_stateroots = match recent_stateroots {
+                    Some(recent_stateroots) => {
+                        let recent_stateroots = recent_stateroots.to_owned();
+                        if recent_stateroots < chain_finality {
+                            return cli_error_and_die(
+                                &format!(
+                                    "\recent-stateroots\" must be greater than {}",
+                                    chain_finality
+                                ),
+                                1,
+                            );
+                        }
+
+                        if recent_stateroots == 0 && *skip_old_messages {
+                            return cli_error_and_die(
+                                "must pass recent-stateroots along with skip-old-messages",
+                                1,
+                            );
+                        }
+
+                        recent_stateroots
+                    }
+                    None => 0,
+                };
+
+                if recent_stateroots == 0 && *skip_old_messages {
+                    return cli_error_and_die(
+                        "Must pass recent stateroots along with skip-old-messages",
+                        1,
+                    );
+                }
+
+                let chain_head = match chain_head().await {
+                    Ok(head) => head.0,
+                    Err(_) => return cli_error_and_die("Could not get network head", 1),
+                };
+
+                let output_path = match output_path {
+                    Some(path) => path.to_owned(),
+                    None => {
+                        let now = OffsetDateTime::now_utc();
+                        format!(
+                            "forest_snapshot_{}_{}_{}_height_{}.car",
+                            now.year(),
+                            now.month(),
+                            now.day(),
+                            chain_head.epoch()
+                        )
+                    }
+                };
+
+                let epoch = tipset.unwrap_or(chain_head.epoch());
+
+                let params = (
+                    epoch,
+                    recent_stateroots,
+                    *skip_old_messages,
+                    output_path.clone(),
+                    TipsetKeysJson(chain_head.key().clone()),
+                );
+
+                let out = chain_export(params)
```

Comment:
Just wondering if we could produce a ```.sha256sum``` checksum file at the same time of creating the snapshot.
This could come up faster that reloading the whole ```.car``` file and hashing it.
But having the hash computed with some command in our DO pipeline will be totally fine (maybe this could be part of #1694).

Reply:
I really like this idea. I just implemented this and it will put the checksum in the same directory as the snapshot

Reply:
As guillaume said, there isn't much point in Forest generating the checksum /after/ the snapshot has been generated. If we cannot do it concurrently then it makes more sense to use an external tool.